src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx

```diff
@@ -118,23 +124,37 @@ export const WorkspacePermissionSettingInput = ({
   );
 
   const handlePermissionModeOptionChange = useCallback(
-    (id: string) => {
-      if (optionIdToWorkspacePermissionModesMap[id]) {
-        onPermissionModesChange([...optionIdToWorkspacePermissionModesMap[id]], index);
+    (option) => {
+      if (option) {
+        const id = option[0].value;
```

Comment:
`if (option)` does not guarantee there is an object inside option, should we add any protection code here?

Reply:
thanks for reminding

Reply:
I guess this `option` can be typed as `Array<EuiComboBoxOptionOption<any>>`?

Reply:
Thanks, Array<EuiComboBoxOptionOption> should be added